assertj-core/src/main/java/org/assertj/core/internal/Classes.java

```diff
@@ -111,6 +115,56 @@ public void assertContainsAnnotations(AssertionInfo info, Class<?> actual,
     if (!missing.isEmpty()) throw failures.failure(info, shouldHaveAnnotations(actual, expected, missing));
   }
 
+  /**
+   * Verifies that permitted subclasses of the actual {@code Class} contain the given {@code Class}es.
+   *
+   * @param info    contains information about the assertion.
+   * @param actual  the "actual" {@code Class}.
+   * @param classes classes that must be permitted subclasses of the actual class
+   * @throws AssertionError if {@code actual} is {@code null}.
+   * @throws AssertionError if the actual {@code Class} does not have all of these permitted subclasses.
+   */
+  public void assertContainsPermittedSubclasses(AssertionInfo info, Class<?> actual,
```

Comment:
This can be pulled to `AbstractClassAssert` as a private method as it's unlikely we would reuse this elsewhere. `hasSuperclass` implementation might be a good example of such a pattern.

Reply:
Ok, will do.

Reply:
Done.